### ISSUE TITLE
Leaf tree node returns true

### ISSUE THREAD

**Issue by caxapexac** [ocornut/imgui#4833]
Version/Branch of Dear ImGui:

Version: 1.86
Branch: docking

Back-end/Renderer/Compiler/OS

Back-ends: webGL
Compiler: em++
Operating System: V8

**My Issue/Question:**

```cpp
        ImGuiTreeNodeFlags flags = ImGuiTreeNodeFlags_Leaf;
        bool opened = ImGui::TreeNodeEx("leaf", flags, "leaf");
        if (opened)
        {
            LOG_WARNING("opened leaf?");
            ImGui::TreePop();
        }
```

Is always opened

Is it by-design?

**Comment by ocornut** [comment]
Yes it always return true. No specific reason but perhaps it would make more sense to return false.
~~You are not supposed to Pop a leaf node either way.~~

**Comment by caxapexac** [comment]
@ocornut should we close this? I think changing of the API could break somebody's code so it would be better to just leave it alone

**Comment by ocornut** [comment]
I'm looking at this now.
FYI this was done in 640c0566 then reverted in 826d77185 but I think the reasoning for revert was incorrect. It's also technically a little bit slower for a feature that's rarely used so I think I'll be changing it back.

**Comment by ocornut** [comment]
I've been circling back and forth about this.... 
I'm tempted to return false. But it would be a slightly breaking change because it means that anyone relying on the implicit `TreePush()` to push aligned items instead a Leaf node would be affected...
For now I added a comment about the current behavior: c234426
